flightsql/src/main/java/io/deephaven/server/flightsql/FlightSqlResolver.java

```diff
@@ -0,0 +1,1916 @@
+            CommandPreparedStatementUpdate.getDescriptor(), CommandPreparedStatementUpdate.class);
+    private static final UnsupportedCommand GET_XDBC_TYPE_INFO_HANDLER =
+            new UnsupportedCommand(CommandGetXdbcTypeInfo.getDescriptor(), CommandGetXdbcTypeInfo.class);
+
+    // Unable to depends on TicketRouter, would be a circular dependency atm (since TicketRouter depends on all the
+    // TicketResolvers).
+    // private final TicketRouter router;
+    private final ScopeTicketResolver scopeTicketResolver;
+    private final ScheduledExecutorService scheduler;
+    private final Authorization authorization;
+    private final KeyedObjectHashMap<ByteString, QueryBase> queries;
+    private final KeyedObjectHashMap<ByteString, PreparedStatement> preparedStatements;
+
+    @Inject
+    public FlightSqlResolver(
+            final AuthorizationProvider authProvider,
+            final ScopeTicketResolver scopeTicketResolver,
+            final ScheduledExecutorService scheduler) {
+        this.authorization = Objects.requireNonNull(authProvider.getTicketResolverAuthorization());
+        this.scopeTicketResolver = Objects.requireNonNull(scopeTicketResolver);
+        this.scheduler = Objects.requireNonNull(scheduler);
+        this.queries = new KeyedObjectHashMap<>(QUERY_KEY);
+        this.preparedStatements = new KeyedObjectHashMap<>(PREPARED_STATEMENT_KEY);
+    }
+
+    /**
+     * The FlightSQL ticket route, equal to {@value FlightSqlTicketHelper#TICKET_PREFIX}.
+     *
+     * @return the FlightSQL ticket route
+     */
+    @Override
+    public byte ticketRoute() {
+        return FlightSqlTicketHelper.TICKET_PREFIX;
+    }
+
+    // ---------------------------------------------------------------------------------------------------------------
+
+    /**
+     * Returns {@code true} if the given command {@code descriptor} appears to be a valid FlightSQL command; that is, it
+     * is parsable as an {@code Any} protobuf message with the type URL prefixed with
+     * {@value FLIGHT_SQL_COMMAND_TYPE_PREFIX}.
+     *
+     * @param descriptor the descriptor
+     * @return {@code true} if the given command appears to be a valid FlightSQL command
+     */
+    @Override
+    public boolean handlesCommand(Flight.FlightDescriptor descriptor) {
+        if (descriptor.getType() != DescriptorType.CMD) {
+            throw new IllegalStateException("descriptor is not a command");
+        }
+        // No good way to check if this is a valid command without parsing to Any first.
+        final Any command = parse(descriptor.getCmd()).orElse(null);
+        return command != null && command.getTypeUrl().startsWith(FLIGHT_SQL_COMMAND_TYPE_PREFIX);
+    }
+
+    // We should probably plumb optional TicketResolver support that allows efficient
+    // io.deephaven.server.arrow.FlightServiceGrpcImpl.getSchema without needing to go through flightInfoFor
```

Comment:
This is an excellent suggestion; file a ticket and add a reference to the ticket in the code? Possibly this comment is more relevant at the call site to `flightInfoFor` when a Schema would be sufficient?

Reply:
I've got a PR for it; it's not a technical requirement since I've got a workaround. I will at a minimum add a link to it from the code. https://github.com/deephaven/deephaven-core/pull/6218